packages/adapter-ccxt/src/CCXTAdapter.ts

```diff
@@ -82,17 +92,28 @@ export class CCXTAdapter implements IAdapter<ExecutionResult, CCXTStep> {
    }
 
    private async initializeExchanges(): Promise<void> {
-      for (const [exchangeId, exchangeConfig] of Object.entries(
-         this.adapterConfig.exchanges
-      ) as Array<[ExchangeId, ApiCredentials]>) {
-         const exchangeService = this.exchangeFactory.createExchangeService(
-            exchangeId,
-            exchangeConfig
-         );
-         this.exchangeServiceRepo.setExchangeService(
-            exchangeId,
-            exchangeService
-         );
-      }
+      const initializationPromises = this.adapterConfig.exchanges.map(
+         async (exchangeConfig) => {
+            if (!exchangeConfig) {
+               return;
+            }
+
+            const { id, apiKey, apiSecret } = exchangeConfig;
+            const exchangeCreds: ApiCredentials = { apiKey, apiSecret };
+
+            const exchangeService =
+               await this.exchangeFactory.createExchangeService(
+                  id as ExchangeId,
+                  exchangeCreds
+               );
+
+            this.exchangeServiceRepo.setExchangeService(
+               id as ExchangeId,
+               exchangeService
+            );
```

Comment:
doesn't this block need to return something in order for you to do `.map()` on `exchangeConfig`?